HPXMLtoOpenStudio/resources/hvac_sizing.rb

```diff
@@ -1886,87 +1886,72 @@ def self.apply_hvac_ground_loop(hvac_sizing_values, weather, hvac_cooling, clima
-      else
-        # Search for first valid bore field
-        new_bore_config = nil
-        valid_configs.keys.each do |bore_config|
-          next unless valid_configs[bore_config].include? num_bore_holes
+    geothermal_loop = hvac_cooling.geothermal_loop
+    bore_spacing = geothermal_loop.bore_spacing
+    bore_diameter = geothermal_loop.bore_diameter
+    grout_conductivity = geothermal_loop.grout_conductivity
+    pipe_r_value = gshp_hx_pipe_rvalue(hvac_cooling)
+    nom_length_heat, nom_length_cool = gshp_hxbore_ft_per_ton(weather, hvac_cooling_ap, bore_spacing, bore_diameter, grout_conductivity, pipe_r_value, climate_zone_iecc)
 
-          new_bore_config = bore_config
-          break
-        end
-        if not new_bore_config.nil?
-          bore_config = new_bore_config
-        else
-          fail 'Could not construct a valid GSHP bore field configuration.'
+    loop_flow = geothermal_loop.loop_flow
+    if loop_flow.nil?
+      loop_flow = [1.0, UnitConversions.convert([hvac_sizing_values.Heat_Capacity, hvac_sizing_values.Cool_Capacity].max, 'Btu/hr', 'ton')].max.floor * 3.0
+    end
+
+    num_bore_holes = geothermal_loop.num_bore_holes
+    if num_bore_holes.nil?
+      num_bore_holes = [1, (UnitConversions.convert(hvac_sizing_values.Cool_Capacity, 'Btu/hr', 'ton') + 0.5).floor].max
+    end
+
+    min_bore_depth = UnitConversions.convert(24.0, 'm', 'ft') # based on g-function library
+    # In NY that's the depth that requires a mining permit, which has been a barrier for Dandelion Energy with installing GSHPs.
+    # Sounds like people are pushing ever deeper but for now we can apply this limit and add a note about where it came from.
+    max_bore_depth = UnitConversions.convert(152.0, 'm', 'ft')
+
+    bore_depth = geothermal_loop.bore_length
+    if bore_depth.nil?
+      bore_length_heat = nom_length_heat * hvac_sizing_values.Heat_Capacity / UnitConversions.convert(1.0, 'ton', 'Btu/hr')
+      bore_length_cool = nom_length_cool * hvac_sizing_values.Cool_Capacity / UnitConversions.convert(1.0, 'ton', 'Btu/hr')
+      bore_length = [bore_length_heat, bore_length_cool].max
+      bore_depth = (bore_length / num_bore_holes).floor # ft
+
+      active_length = 5 # the active length starts about 5 ft below the surface
+      for _i in 0..4
+        if (bore_depth + active_length < min_bore_depth) && (num_bore_holes > 1)
+          num_bore_holes -= 1
+          bore_depth = (bore_length / num_bore_holes).floor
+        elsif bore_depth + active_length > max_bore_depth
+          num_bore_holes += 1
+          bore_depth = (bore_length / num_bore_holes).floor
         end
       end
+
+      bore_depth = (bore_length / num_bore_holes).floor + active_length
     end
 
-    spacing_to_depth_ratio = bore_spacing / bore_depth
+    if (bore_depth < min_bore_depth) || (bore_depth > max_bore_depth)
+      fail "Bore depth (#{bore_depth}) not between #{min_bore_depth.round(3)} and #{max_bore_depth.round(3)} ft."
```

Comment:
There's a routine to "help" autosized borefield not hit this, although I'm not sure if it's guaranteed.

Reply:
Right. I'm asking if you can check that the autosized routine enforces these limits. It would be a bad outcome if the user gets an error from the autosizing routine.

Reply:
I'm refactored the number/depth boreholes portion of the method. Now, if you settle on a number of boreholes leading to average depth that is out of range, the depth is then set to the max (or min) and issues a warning (not an error).

Reply:
If it autosizes to 4 boreholes at 625ft, we don't want to simulate 4 boreholes at 500ft (assuming that's the max), rather we should simulate 5 boreholes at 500ft (and no warning is needed). Or am I misunderstanding your comment?

Reply:
Yes, it would simulate the latter. Sorry, I didn't do a good job explaining. By "settle", I mean hit the 10 (or 1) max (or min) number of boreholes.